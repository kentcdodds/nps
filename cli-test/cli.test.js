import {resolve} from 'path'
import runNPS from './run-nps'

const fixturesPath = resolve(__dirname, './fixtures')

test(
  'without arguments',
  () => snapshot()
)

test(
  'with config with default script',
  () => snapshot('-c ./package-scripts-with-default.js')
)

test(
  'with a missing config',
  () => snapshot('-c ./something-that-does-not-exist.js')
)

test(
  'with --silent',
  () => snapshot('test --silent')
)

test(
  'with --require',
  () => snapshot('--config ./es6-package-scripts.js --require babel-register log')
)

function snapshot(args) {
  return runNPS(fixturesPath, args).then(results => {
    const snapshottableResults = relativizePaths(results)
    expect(snapshottableResults).toMatchSnapshot()
  })
}

/**
 * This takes the results object and removes environment-specific elements from the path.
 * @param {Object} results - This is the results object from runNPS
 * @return {Object} - The new results object with the clean paths
 */
function relativizePaths(results) {
  return Object.keys(results).reduce((obj, key) => {
    obj[key] = results[key].replace(resolve(__dirname, '../'), '<projectRootDir>')
    return obj
  }, {})
}
