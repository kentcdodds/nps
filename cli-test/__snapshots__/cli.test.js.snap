exports[`test with --require 1`] = `
Object {
  "stderr": "",
  "stdout": "nps executing: echo \"log\"
log
"
}
`;

exports[`test with --silent 1`] = `
Object {
  "stderr": "",
  "stdout": "test script
"
}
`;

exports[`test with a missing config 1`] = `
Object {
  "stderr": "Unable to find JS config at \"./something-that-does-not-exist.js\". Attempted to require as \"<projectRootDir>/cli-test/fixtures/something-that-does-not-exist.js\" https://github.com/kentcdodds/p-s/blob/v0.0.0-semantically-released/other/ERRORS_AND_WARNINGS.md#unable-to-find-config
",
  "stdout": ""
}
`;

exports[`test with config with default script 1`] = `
Object {
  "stderr": "",
  "stdout": "nps executing: echo \"default script\"
default script
"
}
`;

exports[`test without arguments 1`] = `
Object {
  "stderr": "",
  "stdout": "
  Usage: nps [options]

  Options:

    -h, --help                                  output usage information
    -V, --version                               output the version number
    -s, --silent                                Silent nps output
    -p, --parallel <script-name1,script-name2>  Scripts to run in parallel (comma seprated)
    -c, --config <filepath>                     Config file to use (defaults to nearest package-scripts.yml or package-scripts.js)
    -l, --log-level <level>                     The log level to use (error, warn, info [default])
    -r, --require <module>                      Module to preload

Available scripts (camel or kebab case accepted)
├─ test - echo \"test script\"
└─ lint - echo \"lint.default\"
   │  └─ lint.sub.thing - this is a description - echo \"deeply nested thing\"
"
}
`;
